### DISCUSSION TITLE
Python sample to create IFC from scratch

### DISCUSSION THREAD

**Discussion by silverhawak59** [IfcOpenShell/IfcOpenShell#4764] [Q&A]
Hi All, 
I recently started using IfcOpenShell and looking for some python sample to create an IFC file from scratch by adding a pipe and  some properties to it.

thanks

**Comment by aothms** [comment]
This one is a bit old, doesn't use any of the modern APIs https://academy.ifcopenshell.org/posts/creating-a-simple-wall-with-property-set-and-quantity-information/

This is likely what you're looking for https://github.com/cvillagrasa/IfcOpenHouse/blob/master/nbs/00_generation.ipynb Thanks to @cvillagrasa

[Speaking of which, @cvillagrasa, shouldn't we also upload that here https://github.com/IfcOpenShell/academy/tree/main/posts]

**Reply by cvillagrasa** [reply]
@aothms absolutely. I'm not having a lot of time lately, but I want to finish the ifcopenshell-jupyter repo, and then I believe the notebook would be ok to be placed on the academy repo.

As a reminder, the two main pending items on the current notebook are:

1. The non-standard usage of the API calls, which now I believe could be changed to the new standard. Is it stable, @Moult ?
2. A reliable viz library. The notebook currently uses the old IFC.js, while the repo is being built around xeokit, which seems more stable.

Lastly, from the list I made on [this comment](https://github.com/IfcOpenShell/ifcopenshell-jupyter/issues/1#issuecomment-1772122003) from October, the main update is that unfortunately I haven't had much time, and then 1) Solved, I just changed it, 2) what do you think about the licensing? 3) Done, but some refactoring still needed. 4) Pending, 5) Pending, not urgent.

**Reply by Moult** [reply]
https://opensource.stackexchange.com/questions/4442/how-does-the-agpl-apply-to-javascript-libraries#4443 I'd say all the JS is AGPL and Python is LGPL. Ping @xeolabs

**Comment by Moult** [comment]
Another resource is https://docs.ifcopenshell.org/ifcopenshell-python/code_examples.html#create-a-simple-model-from-scratch

For a pipe, you can use a profile https://docs.ifcopenshell.org/ifcopenshell-python/geometry_creation.html#profile-representations or swept disk https://docs.ifcopenshell.org/ifcopenshell-python/geometry_creation.html#custom-representations

Then add pset https://docs.ifcopenshell.org/autoapi/ifcopenshell/api/pset/index.html#ifcopenshell.api.pset.edit_pset
